packages/functional-tests/src/runner/test-runner.ts

```diff
@@ -2,72 +2,87 @@
 import { getDefinedTestsMetadata } from '../test-decorator';
 
 // tslint:disable: no-any no-unsafe-any
 
 export interface TestRunLogProperties {
-    source: LogSource;
+    runId: string;
+    releaseId: string;
+    environment: string;
     testContainer: string;
     testName: string;
-    environment: string;
     result: TestResult;
     error?: string;
 }
 
 export interface TestContainerLogProperties {
-    source: LogSource;
-    testContainer: string;
+    runId: string;
+    releaseId: string;
     environment: string;
+    testContainer: string;
     result: TestResult;
 }
 
 export class TestRunner {
-    public constructor(@inject(Logger) private readonly logger: Logger) {}
+    public constructor(
+        @inject(Logger) private readonly logger: Logger,
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+    ) {}
+
+    public async runAll(testContainers: object[], env: TestEnvironment, releaseId: string, runId?: string): Promise<void> {
+        const currentRunId = runId !== undefined ? runId : this.guidGenerator.createGuid();
+        await Promise.all(testContainers.map(async testContainer => this.run(testContainer, env, releaseId, currentRunId)));
+    }
 
-    public async run(testContainer: object, env: TestEnvironment): Promise<void> {
+    public async run(testContainer: object, env: TestEnvironment, releaseId: string, runId?: string): Promise<void> {
+        const currentRunId = runId !== undefined ? runId : this.guidGenerator.createGuid();
         const definedTests = getDefinedTestsMetadata(testContainer);
         // tslint:disable-next-line: no-bitwise
         const targetedTests = definedTests.filter(definedTest => definedTest.environments & env);
         let containerPass = true;
         await Promise.all(
             targetedTests.map(async targetedTest => {
-                const testPass = await this.runTest(targetedTest, env);
+                const testPass = await this.runTest(targetedTest, env, releaseId, currentRunId);
                 containerPass = containerPass && testPass;
             }),
         );
 
         const testContainerName = testContainer.constructor.name;
-        this.log(`[E2E] Test container ${testContainerName} ${containerPass ? 'pass' : 'fail'}`, LogLevel.info, {
-            source: 'e2e',
-            testContainer: testContainerName,
+        this.log({
+            runId: currentRunId,
+            releaseId: releaseId,
             environment: TestEnvironment[env],
+            testContainer: testContainerName,
             result: containerPass ? 'pass' : 'fail',
         });
     }
 
-    private async runTest(testDefinition: TestDefinition, env: TestEnvironment): Promise<boolean> {
+    private async runTest(testDefinition: TestDefinition, env: TestEnvironment, releaseId: string, runId: string): Promise<boolean> {
         try {
             await Promise.resolve(testDefinition.testImplFunc());
```

Comment:
Kelsey and I found that this function need to be bind with the test container instance, that's included in her PR.